src/Azure.Functions.Cli/Helpers/PythonHelpers.cs

```diff
@@ -73,85 +76,100 @@ private static void CreateRequirements()
             }
         }
 
-        public static async Task<WorkerLanguageVersionInfo> ValidatePythonVersion(
-            bool setWorkerExecutable = false,
-            bool errorIfNoExactMatch = false,
-            bool errorOutIfOld = true)
+        public static void AssertPythonVersion(WorkerLanguageVersionInfo pythonVersion, bool errorIfNotSupported = false, bool errorIfNoVersion = true)
         {
-            // If users are overriding this value, we don't have to worry about verification
+            if (pythonVersion?.Version == null)
+            {
+                var message = "Could not find a Python version. Python 3.6.x or 3.7.x is recommended, and used in Azure Functions.";
+                if (errorIfNoVersion) throw new CliException(message);
+                ColoredConsole.WriteLine(WarningColor(message));
+                return;
+            }
+
+            ColoredConsole.WriteLine(AdditionalInfoColor($"Found Python version {pythonVersion.Version} ({pythonVersion.ExecutablePath})."));
```

Comment:
I think the proper place to report the python version is in **GetEnvironmentPythonVersion**.
Since we had a cache there, we only need to report the python version when we doing the first detection.
Any thoughts?

Reply:
That's fair. But I think when asserting is really when it matters what version users have. So, say if you don't have 3.7, the output would be --

```
Found Python version {pythonVersion.Version} ({pythonVersion.ExecutablePath}).
Python 3.6.x or 3.7.x is required for this operation.
Please install Python 3.6 or 3.7, and use a virtual environment to switch to Python 3.6 or 3.7.
```

Reply:
Hmm, you made a good point. Yea, let's keep this line here.